### DISCUSSION TITLE
Handling exceptions while using file-based routing

### DISCUSSION THREAD

**Discussion by yoonminsang** [TanStack/router#2489] [Ideas]
I'm using file based routing. But I want to create a non-router file in the routes folder. For example, if there are many subcomponents or hooks in a file called signup.tsx, it is natural to separate the files.

Of course, there is also a way to create a component in a folder like pages and connect it to the routes folder. However, I don't want codegen to be automatically activated when folders like _except, _compoennts, and _hooks are added to routes.

How about a function that blocks codegen for files in a specific folder in the plugin? I'm curious about your thoughts

**Comment by schiller-manuel** [comment]
you can configure this via either `routeFileIgnorePrefix` or `routeFileIgnorePattern`

see https://tanstack.com/router/latest/docs/framework/react/guide/file-based-routing#options

**Reply by yoonminsang** [reply]
Thank you
